autoload/codi.vim

```diff
@@ -186,6 +186,22 @@ function! s:user_au(au)
   endif
 endfunction
 
+" If percent / float width, calculate based on buf width
+" while respecting |winwidth|
+" Else, return absolute width as given
+function! s:pane_width()
+  let width = s:get_opt('width')
+
+  if type(width) == type(0)
+    return width
+  endif
+
+  let buf_width  = winwidth(bufwinnr('%'))
+  let pane_width = buf_width * max([0, float2nr(ceil(width))]) / 100
```

Comment:
I see what you mean by lost precision now...

I think you were correct in using the ternary earlier. Changing the line to the following should allow us to keep precision.

```vim
  let pane_width = float2nr(ceil(buf_width * (width < 0.0 ? 0.0 : width) / 100))
```

Reply:
Actually, the `round` function may be even better:

```vim
  let pane_width = float2nr(round(buf_width * (width < 0.0 ? 0.0 : width) / 100))
```

Reply:
When I get home I will start on these changes. There is no space on the train to allow for coding at the moment 😅